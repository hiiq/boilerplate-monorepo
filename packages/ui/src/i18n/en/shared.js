const languages = {
  dev: '𝐹åḱë',
  en: 'English',
};

export const shared = {
  toggles: {
    availableLanguages: 'Available Languages',
    chooseALanguage: 'Choose a language',
    languages,
    layout: 'Layout',
    modes: {
      dark: 'Set app theme to Dark Mode',
      light: 'Set app theme to Light Mode',
    },
    theme: 'Theme',
    ...languages,
  },
};
