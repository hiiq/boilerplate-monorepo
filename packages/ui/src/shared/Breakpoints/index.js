import { useMediaQuery } from 'react-responsive';

const Breakpoint = {
  HUGE: 1440,
  LARGE: 1170,
  MEDIUM: 768,
  SMALL: 450,
};

const Huge = ({ children }) => {
  const isHuge = useMediaQuery({ minWidth: Breakpoint.HUGE });

  return isHuge ? children : null;
};

const Desktop = ({ children }) => {
  const isDesktop = useMediaQuery({ minWidth: Breakpoint.LARGE });

  return isDesktop ? children : null;
};

const Tablet = ({ children }) => {
  const isTablet = useMediaQuery({ minWidth: Breakpoint.MEDIUM });

  return isTablet ? children : null;
};

const Mobile = ({ children }) => {
  const isMobile = useMediaQuery({ maxWidth: Breakpoint.SMALL - 1 });

  return isMobile ? children : null;
};

const GreaterThanMobile = ({ children }) => {
  const isMobile = useMediaQuery({ minWidth: Breakpoint.SMALL });

  return isMobile ? children : null;
};

const Default = ({ children }) => {
  const isNotMobile = useMediaQuery({ minWidth: 0 });

  return isNotMobile ? children : null;
};

export { Default, Desktop, GreaterThanMobile, Huge, Mobile, Tablet };
