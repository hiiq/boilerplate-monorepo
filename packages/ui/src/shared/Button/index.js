import { Utils } from '@boilerplate-monorepo/common';
import { A11y } from '@boilerplate-monorepo/ui-common';
import { bool, func, node, string } from 'prop-types';
import { path } from 'ramda';
import React, { useEffect, useRef, useState } from 'react';
import { useFormContext } from 'react-hook-form';
import { animated, useSpring } from 'react-spring';
import styled, { css } from 'styled-components/macro';
import { CustomProperty } from 'types/customProperties';
import { Loader } from '../Loader';
import { Context } from './context';
import { Size } from './size';
import { Style } from './style';
import { Type } from './type';

const { Role } = A11y;

const styles = css`
  align-items: center;
  border-radius: 0.125rem;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: grid;
  justify-items: center;
  opacity: 0.95;
  position: relative;
  text-align: center;
  transition: opacity ${CustomProperty.TRANSITION_DELAY} ${
  CustomProperty.TRANSITION_TIMING_FUNCTION
};
  vertical-align: middle;
  width: 100%;

  /* stylelint-disable-next-line order/properties-alphabetical-order */
  ${Style.context}
  ${Style.size}

  &:active,
  &:focus,
  &:hover {
    opacity: 1;
  }

  /* stylelint-disable-next-line order/properties-alphabetical-order */
  ${({ disabled }) =>
    disabled &&
    css`
      cursor: default;
      opacity: 0.7;
      pointer-events: none;

      &:active,
      &:focus,
      &:hover {
        cursor: default;
        opacity: 0.7;
        pointer-events: none;
      }
    `}

  ${({ isAutoWidth }) =>
    isAutoWidth &&
    css`
      width: fit-content;
    `}

  /* stylelint-disable-next-line order/properties-alphabetical-order */
  ${({ className }) => className}

  ${({ height, width }) =>
    height &&
    width &&
    css`
      height: ${height}px;
      width: ${width}px;
    `}
`;

const StyledButton = styled.button`
  ${styles}
`;

const StyledFancyLoader = animated(styled.div`
  display: grid;
  position: relative;
`);

const FancyLoader = fadeStyles => (
  <StyledFancyLoader style={fadeStyles}>
    <Loader />
  </StyledFancyLoader>
);

const StyledFancyContent = animated(styled.div`
  display: grid;
  position: relative;
`);

const FancyContent = ({ children, ...fadeStyles }) => (
  <StyledFancyContent style={fadeStyles}>{children}</StyledFancyContent>
);

FancyContent.propTypes = {
  children: node.isRequired,
};

// eslint-disable-next-line complexity, max-statements
const Button = ({
  children,
  className,
  context,
  disabled,
  isAutoWidth,
  label,
  onClick,
  text,
  type,
  ...props
}) => {
  const [width, setWidth] = useState(0);
  const [height, setHeight] = useState(0);
  const formProps = useFormContext();
  const isSubmitting = path(['formState', 'isSubmitting'], formProps);
  const [isLoaderVisible, beLoaderVisible] = useState(isSubmitting);
  const fadeOutProps = useSpring({ opacity: isLoaderVisible ? 1 : 0 });
  const fadeInProps = useSpring({ opacity: isLoaderVisible ? 0 : 1 });
  const ref = useRef(null);

  const content = children || text;

  useEffect(() => {
    if (!ref.current) return;

    const {
      height: rectHeight,
      width: rectWidth,
    } = ref.current.getBoundingClientRect();

    if (rectHeight) {
      setHeight(rectHeight);
    }

    if (rectWidth) {
      setWidth(rectWidth);
    }
  }, [content, isSubmitting]);

  useEffect(() => {
    if (isSubmitting) {
      beLoaderVisible(true);
    }

    // Show loader a bits longer to avoid loading flash
    if (!isSubmitting && isLoaderVisible) {
      const timeout = setTimeout(() => beLoaderVisible(false), 400);

      // Donâ€™t forget to clear the timeout
      return () => clearTimeout(timeout);
    }

    return () => null;
  }, [isLoaderVisible, isSubmitting]);

  const isDisabled = disabled || isLoaderVisible;

  return (
    <StyledButton
      {...props}
      aria-disabled={isDisabled}
      aria-label={label || text}
      className={className}
      context={context}
      disabled={isDisabled}
      height={height}
      isAutoWidth={isAutoWidth}
      onClick={onClick}
      ref={ref}
      role={Role.BUTTON}
      type={type}
      width={width}
    >
      {isLoaderVisible ? (
        <FancyLoader {...fadeOutProps} />
      ) : (
        <FancyContent {...fadeInProps}>{content}</FancyContent>
      )}
    </StyledButton>
  );
};

Button.defaultProps = {
  children: undefined,
  className: undefined,
  context: Context.DEFAULT,
  disabled: false,
  isAutoWidth: false,
  isSubmitting: undefined,
  label: undefined,
  onClick: Utils.noop,
  size: Size.DEFAULT,
  type: Type.BUTTON,
};

Button.propTypes = {
  children: node,
  className: string,
  context: Context.propTypes,
  disabled: bool,
  isAutoWidth: bool,
  isSubmitting: bool,
  label: ({ label, text }) => {
    if (text || label) return undefined;

    return new Error('The `label` prop is required when no text is set');
  },
  onClick: func,
  size: Size.propTypes,
  text: ({ children, text }) => {
    if (children || text) return undefined;

    return new Error(
      `The 'text' prop is required when no children are provided`
    );
  },
  type: Type.propTypes,
};

export { Button, Context, Size, Type, styles };
