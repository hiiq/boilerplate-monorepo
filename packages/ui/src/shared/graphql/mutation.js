import { gql } from '@apollo/client';

const USER_LOGIN = gql`
  mutation userLogin($input: UserLoginInput!) {
    userLogin(input: $input)
  }
`;

const USER_LOGOUT = gql`
  mutation userLogout {
    userLogout
  }
`;

const USER_REGISTER = gql`
  mutation userRegister($input: UserRegisterInput!) {
    userRegister(input: $input)
  }
`;

export const Mutation = {
  USER_LOGIN,
  USER_LOGOUT,
  USER_REGISTER,
};
