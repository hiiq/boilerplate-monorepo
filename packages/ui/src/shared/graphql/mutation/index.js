export * from './useUserLogin';
export * from './useUserLogout';
export * from './useUserRegister';
export * from './useUserSelfUpdate';
