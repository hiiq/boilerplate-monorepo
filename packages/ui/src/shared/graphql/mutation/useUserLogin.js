import { gql } from '@apollo/client';
import { useMutation } from '../useMutation';

const MUTATION_USER_LOGIN = gql`
  mutation userLogin($input: UserLoginInput!) {
    userLogin(input: $input)
  }
`;

const useUserLogin = options => useMutation(MUTATION_USER_LOGIN, options);

export { useUserLogin };
