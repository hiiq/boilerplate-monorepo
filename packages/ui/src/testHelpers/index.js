import { cleanup, render } from '@testing-library/react';
import { curry } from 'ramda';
import React from 'react';
import { Providers } from 'modules/app';

export * from '@testing-library/react';

const customRender = curry((Component, defaultProps = {}, overrides = {}) =>
  render(<Component {...defaultProps} {...overrides} />, {
    wrapper: Providers,
  })
);

export { cleanup, customRender as render };
