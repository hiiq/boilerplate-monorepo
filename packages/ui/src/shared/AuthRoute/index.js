import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { useAuth } from 'shared/useAuth';
import { Route as RouteType } from 'types/route';

const AuthRoute = props => {
  const { isAuthenticated } = useAuth();

  if (!isAuthenticated) {
    return <Redirect to={RouteType.DASHBOARD.path} />;
  }

  return <Route {...props} />;
};

export { AuthRoute };
