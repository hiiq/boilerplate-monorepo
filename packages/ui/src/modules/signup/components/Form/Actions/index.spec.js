import React from 'react';
import { QUERY_USER_SELF } from 'shared/graphql';
import * as MutationHooks from 'shared/graphql/mutation/useUserLogin';
import { render } from 'testHelpers';
import { Actions } from '.';

jest.mock('shared/Link');
jest.mock('shared/forms/SubmitButton');

describe('<Actions/>', () => {
  const defaultProps = {};

  const renderComponent = (overrides) =>
    render(<Actions {...defaultProps} {...overrides} />);

  beforeEach(() => {
    const useUserLogin = td.replace(MutationHooks, 'useUserLogin');

    td.when(
      useUserLogin({
        refetchQueries: [{ query: QUERY_USER_SELF }],
      })
    ).thenReturn([jest.fn().mockName('userLogin'), { loading: false }]);
  });

  test('renders properly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
