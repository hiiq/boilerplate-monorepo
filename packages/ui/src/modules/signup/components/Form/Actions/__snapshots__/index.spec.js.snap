// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Actions/> renders properly 1`] = `
<div>
  <div
    class="Actions__Container-jCrrLB hwjhEZ"
  >
    <x-submitbutton
      isautowidth="true"
      text="t(signup)"
    />
    <div />
    <x-link
      data-testid="link"
      to="/login"
    >
      t(login)
    </x-link>
  </div>
</div>
`;
